### PR TITLE
resolves #2743 allowing to increment env var

### LINKED ISSUE #2743: Allow to increment a env var for instances of an application started
For example i want to run my application in fork_mode with 4 process, but only one declaration.

```
{
    name       : "api-app",
    script     : "./api.js",
    instances  : 4,
    exec_mode  : "fork"
    env: {
      PORT: 3000
    }
  }
```

And add something like that : 
```
{
    name       : "api-app",
    script     : "./api.js",
    instances  : 4,
    increment_foreach: 'PORT',
    exec_mode  : "fork"
    env: {
      PORT: 3000
    }
  }
```

### PR DESCRIPTION
**Pull request by btmdave**
Please always submit pull requests on the development branch.

| Q             | A
| ------------- | ---
| Bug fix?      |no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #2743
| License       | MIT
| Doc PR        | https://github.com/pm2-hive/pm2-hive.github.io/pulls

*Please update this template with something that matches your PR*

### CODE REVIEW

File: lib/God.js [hunk 1]
```diff
@@ -362,25 +362,30 @@ God.prepare = function prepare(env, cb) {
...
 
     (function ex(i) {
       if (i <= 0) {
         if (cb) return cb(null, arr);
         return false;
       }
 
-      env.NODE_APP_INSTANCE = instance_id++;
-      env.vizion_running = false;
+      if (env.en && env.env.increment_foreach)
```
Comment (soyuka) [inline lib/God.js]: `env.en`?

File: lib/God.js [hunk 2]
```diff
@@ -362,25 +362,30 @@ God.prepare = function prepare(env, cb) {
...
+        env[env.env.increment_foreach] = increment++;
```
Comment (soyuka) [inline lib/God.js]: @vmarchaud I assume `increment_foreach` is a new pm2 option. Therefore, I'd like to propose `increment` instead.

Btw we need to add this to the schema and the docs too!

**Comment by vmarchaud** [comment]
Yeah, i suggested the modification but looks like @btmdave just copy-pasted what i wrote here, i'll certainly make a commit into dev branch myself.

**Comment by btmdave** [comment]
thanks @vmarchaud I did copy what you had put, I wasn't sure how to test this locally here.  Is there a way to test a deployment w/ a dev branch?

**Comment by vmarchaud** [comment]
@btmdave I meat that it need to have few units tests (you can look [here](https://github.com/Unitech/pm2/tree/master/test/bash) for inspiration)

File: lib/API/schema.json [hunk 3]
```diff
@@ -182,5 +182,8 @@
     "type": [
       "boolean"
     ]
+  },
+  "increment": {
+    "type": "number"
```
Comment (soyuka) [inline lib/API/schema.json]: This is wrong, `increment` will be a `string` which is an environment `key` to be incremented

File: lib/God.js [hunk 4]
```diff
@@ -362,25 +362,30 @@ God.prepare = function prepare(env, cb) {
...
     (function ex(i) {
       if (i <= 0) {
         if (cb) return cb(null, arr);
         return false;
       }
 
-      env.NODE_APP_INSTANCE = instance_id++;
-      env.vizion_running = false;
+      if (env.env && env.env.increment)
+        env[env.env.increment] = increment++;
```
Comment (soyuka) [inline lib/God.js]: What about:

```
env[env.env.increment] = !env[env.env.increment] ? 1 : env[env.env.increment] + 1
```

We don't really need a new variable for this.

**Comment by vmarchaud** [comment]
Implemented via https://github.com/Unitech/pm2/commit/35945e82d0a67c2368b4ab17facb96da612e0e2f

**Pull request closed without merging**